src/XdgShell.cpp

```diff
@@ -1,67 +1,50 @@
 #include "XdgShell.hpp"
 #include "Server.hpp"
 
-namespace XdgShell
-{
-  void xdg_surface_map(struct wl_listener *listener, void *data)
-  {
-    View *view = wl_container_of(listener, view, map);
-    view->mapped = true;
-    ServerView::focus_view(view, view->xdg_surface->surface);
-  };
-
-  void xdg_surface_unmap(struct wl_listener *listener, void *data)
-  {
-    View *view = wl_container_of(listener, view, unmap);
-    view->mapped = false;
-  };
+XdgShell::XdgShell(Server *server, struct wl_display *display) : server(server) {
+  xdg_shell = wlr_xdg_shell_create(display);
+  SET_LISTENER(XdgShell, XdgShellListeners, new_xdg_surface, server_new_xdg_surface);
+  wl_signal_add(&xdg_shell->events.new_surface, &new_xdg_surface);
+}
 
-  void xdg_surface_destroy(struct wl_listener *listener, void *data)
-  {
-    View *view = wl_container_of(listener, view, destroy);
-    wl_list_remove(&view->link);
-    free(view);
-  };
+void XdgShell::xdg_surface_map(struct wl_listener *listener, void *data)
+{
+  view->mapped = true;
+  ServerView::focus_view(view, view->xdg_surface->surface);
+};
 
-  void xdg_toplevel_request_move(struct wl_listener *listener, void *data)
-  {
-    View *view = wl_container_of(listener, view, request_move);
-    ServerView::begin_interactive(view, CursorMode::CURSOR_MOVE, 0);
-  };
+void XdgShell::xdg_surface_unmap(struct wl_listener *listener, void *data)
+{
+  view->mapped = false;
+};
 
-  void xdg_toplevel_request_resize(struct wl_listener *listener, void *data)
-  {
-    struct wlr_xdg_toplevel_resize_event *event = static_cast<struct wlr_xdg_toplevel_resize_event *>(data);
-    View *view = wl_container_of(listener, view, request_resize);
-    ServerView::begin_interactive(view, CursorMode::CURSOR_RESIZE, event->edges);
-  };
+void XdgShell::xdg_surface_destroy(struct wl_listener *listener, void *data)
+{
+  wl_list_remove(&view->link);
+  //Maybe delete the view here ?
```

Comment:
```suggestion
  // TODO: Maybe delete the view here ?
```
Also please open an issue or put a note somewhere so we don't forget this.